include/pqxx/field.hxx

```diff
@@ -243,13 +243,30 @@ public:
     return as<O<T>>();
   }
 
+  /// Read SQL array contents as a @ref pqxx::array.
+  template<typename ELEMENT, auto... ARGS>
+  array<ELEMENT, ARGS...> as_sql_array() const
+  {
+    using array_type = array<ELEMENT, ARGS...>;
+
+    // There's no such thing as a null SQL array.
+    if (is_null())
+      internal::throw_null_conversion(type_name<array_type>);
+    else
+      return array_type{this->view(), this->m_home.m_encoding};
+  }
+
   /// Parse the field as an SQL array.
   /** Call the parser to retrieve values (and structure) from the array.
    *
    * Make sure the @ref result object stays alive until parsing is finished. If
    * you keep the @ref row of `field` object alive, it will keep the @ref
    * result object alive as well.
    */
+  [[deprecated(
```

Comment:
I'm not sure about this deprecation. Looks like pqxx::array doesn't allow move and extraction to an external structure, which could be useful. Otherwise looks greate

Reply:
Thanks.  I suppose we could consider supporting moves, if we find a good use for them.  But when are they really worthwhile?  Perhaps if the array elements were of a custom time which in turn contained an array, or something like that.

Same consideration for extraction to an external structure: when is that helpful?  You can copy elements right now.  Or just keep the `pqxx::array` object as the basic storage for that data.

Reply:
@fallenworld1 for now if you don't mind I think I'll just merge this, and then make any incremental improvements as needed.